cmd/zc_traverser_blob.go

```diff
@@ -236,13 +239,83 @@ func (t *blobTraverser) Traverse(preprocessor objectMorpher, processor objectPro
 	return t.serialList(containerURL, blobUrlParts.ContainerName, searchPrefix, extraSearchPrefix, preprocessor, processor, filters)
 }
 
+// HasDirectoryChangedSinceLastSync checks directory changed since lastSyncTime.
+// It honours CFDMode to make the decision, f.e., if CFDMode allows ctime/mtime to be used for CFD it may not need to query
+// attributes from target. If it returns True, TargetTraverser will enumerate the directory and compare each enumerated object
+// with the source scanned objects in ObjectIndexer[] to find out if the object needs to be sync'ed.
+//
+// Note: For CFDModes that allow ctime for CFD we can avoid enumerating target dir if we know directory has not changed.
+//       This increases sync efficiency.
+//
+// Note: If we discover that certain sources cannot be safely trusted for ctime update we can change this to return True for them,
+//       thus falling back on the more rigorous target<->source comparison.
+func (t *blobTraverser) HasDirectoryChangedSinceLastSync(so StoredObject, containerURL azblob.ContainerURL) bool {
+	// CFDMode is to enumerate Target and compare. So we return true in that case.
+	// This will lead to enumeration of target.
```

Comment:
the comments in the doc are more appropriate, pls use them as-is.
f.e. here the following comment is apt

	        //
	        // Force enumeration for TargetCompare mode.
	        // For other CFDModes we enumerate a directory iff it has changed
	        // since last sync.
	        //